File: Python/Punkty w okregu.py
xo, yo, r = map(int, input().split())
n = int(input())
rSquared = r * r
for _ in range(n):
    xn, yn = map(int, input().split())
    dx = xn - xo
    dy = yn - yo
    dist_squared = dx * dx + dy * dy
    if dist_squared < rSquared:
        print("I")
    elif dist_squared > rSquared:
        print("O")
    else:
        print("E")

File: Java/Punkty w okregu.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int xo = input.nextInt();
        int yo = input.nextInt();
        int r = input.nextInt();
        int n = input.nextInt();
        int rsquared = r * r;
        for (int i = 0; i < n; i++) 
        {
            int xi = input.nextInt();
            int yi = input.nextInt();
            int dx = xi - xo;
            int dy = yi - yo;
            int distSquared = dx * dx + dy * dy;
            if (distSquared < rsquared)
                System.out.println("I");
            else if (distSquared > rsquared)
                System.out.println("O");
            else
                System.out.println("E");
        }
        input.close();
    }
}